src/vdom/diff.js

```diff
@@ -54,9 +54,17 @@ export function diff(dom, vnode, context, mountAll, parent, componentRoot) {
 	}
 
 	let ret = idiff(dom, vnode, context, mountAll, componentRoot);
-
-	// append the element if its a new parent
-	if (parent && ret.parentNode!==parent) parent.appendChild(ret);
+	if (parent) {
+		// append the element if its a new parent
+		if (ret instanceof Array) {
```

Comment:
https://jsperf.com/instanceof-array-vs-array-isarray/33

can you try `ret.constructor === Array`?
its seems faster on many browsers
(you have two other instances of this in this file)